test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -189,6 +190,31 @@
                         "supported"]
                        "bogus"}]})))))))
 
+(deftest valid-static-file-path-test
+  (let [valid-paths ["modules/foo/files/bar.txt"
+                     "modules/foo/files/bar"
+                     "modules/foo/files/bar/baz.txt"
+                     "modules/foo/files/bar/more/path/elements/baz.txt"]
+        invalid-paths ["modules/foo/manifests/bar.pp"
+                       "manifests/site.pp"
+                       "environments/foo/bar/files"
+                       "environments/../manifests/files/site.pp"
+                       "environments/../modules/foo/lib/puppet/parser/functions/site.rb"
+                       "environments/production/files/~/.bash_profile"
+                       "environments/../modules/foo/files/site.pp"
+                       "environments/production/modules/foo/files/../../../../../../site.pp"
+                       "environments/production/modules/foo/files/bar.txt"
+                       "environments/production/modules/foo/files/..conf..d../..bar..txt.."
+                       "environments/test/modules/foo/files/bar/baz.txt"
+                       "environments/dev/modules/foo/files/path/to/file/something.txt"]
```

Comment:
@haus I just realized that @joshcooper is suggesting here that we reject this kind of path, but the test case you added in c1dee2fd82e42b1aba2db86ab88b6df40a35f997 seems to show that we allow it and pass it along to `code-id-command`.

... or am I still confused?